content/guides/end-to-end-testing/testing-your-app.md

```diff
@@ -410,111 +410,87 @@ You'll likely also want to test your login UI for:
+  // UI should reflect this user being logged in
+  cy.get('h1').should('contain', username)
+})
 
-Read about [best practices](/guides/references/best-practices) here.
+// In your spec file
 
-</Alert>
+it('does something on a secured page', function () {
+  const { username, password } = this.currentUser
+  cy.login(username, password)
 
-Using your UI to **log in** is the _exact same scenario_ as what we described
-previously. Logging in is a prerequisite of state that comes before all of your
-other tests.
+  // ...rest of test
+})
+```
 
-Because Cypress isn't Selenium, we can actually take a huge shortcut here and
-skip needing to use our UI by using [`cy.request()`](/api/commands/request).
+#### Improving performance
 
-Because [`cy.request()`](/api/commands/request) automatically gets and sets
-cookies under the hood, we can actually use it to build up state without using
-your browser's UI, yet still have it perform exactly as if it came from the
-browser!
+You're probably wondering what happened to our advice about logging in "only
+once". The custom command above will work just fine for testing your secured
+pages, but if you have more than a handful of tests, logging in before every
+test is going to add a lot of running time to your suite.
 
-Let's revisit the example from above but assume we're testing some other part of
-the system.
+Luckily, Cypress provides the [`cy.session()` command](/api/commands/session), a
+powerful performance tool that lets you cache user session data and reuse it for
+multiple tests without going through multiple login flows! Let's modify the
+`cy.login()` command from our previous example to use `cy.session()`:
 
 ```js
-describe('The Dashboard Page', () => {
-  beforeEach(() => {
-    // reset and seed the database prior to every test
-    cy.exec('npm run db:reset && npm run db:seed')
-
-    // seed a user in the DB that we can control from our tests
-    // assuming it generates a random password for us
-    cy.request('POST', '/test/seed/user', { username: 'jane.lane' })
-      .its('body')
-      .as('currentUser')
-  })
-
-  it('logs in programmatically without using the UI', function () {
-    // destructuring assignment of the this.currentUser object
-    const { username, password } = this.currentUser
-
-    // programmatically log us in without needing the UI
-    cy.request('POST', '/login', {
-      username,
-      password,
-    })
-
-    // now that we're logged in, we can visit
-    // any kind of restricted route!
-    cy.visit('/dashboard')
-
-    // our auth cookie should be present
-    cy.getCookie('your-session-cookie').should('exist')
-
-    // UI should reflect this user being logged in
-    cy.get('h1').should('contain', 'jane.lane')
-  })
+Cypress.Commands.add('login', (username, password) => {
+  cy.session(
+    username,
+    () => {
+      cy.visit('/login')
+      cy.get('input[name=username]').type(username)
+      cy.get('input[name=password]').type(`${password}{enter}`)
```

Comment:
```suggestion
      cy.get('input[name=password]').type(`${password}{enter}`, { log: false }) // use {log: false} to prevent your password from showing in the Command Log 
```